src/transformStateWithClones.js

```diff
@@ -6,8 +6,32 @@
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+
+function transformStateWithClones(initialState, actions) {
+  const states = [];
+  let currentState = { ...initialState };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      const properties = action.extraData;
```

Comment:
Consider adding a check to ensure `action.extraData` is an object before iterating over it. This will prevent potential runtime errors if `extraData` is not provided or is not an object.